### PR TITLE
record submission lag as metric in datadog

### PR DESCRIPTION
**Pull request by snopoke**

### CODE REVIEW

**Review by proteusvacuum — APPROVED** [review]

File: corehq/apps/receiverwrapper/views.py [hunk 1]
```diff
@@ -160,13 +160,17 @@ def _submission_error(request, message, count_metric, metric_tags,
     return response
 
 
-def _record_metrics(tags, submission_type, response, timer=None):
+def _record_metrics(tags, result, timer=None):
+    if result.xform and result.xform.metadata:
```
Comment (emord) [inline corehq/apps/receiverwrapper/views.py]:
```suggestion
    if hasattr(result, 'xform') and result.xform.metadata:
```

Comment (emord) [reply corehq/apps/receiverwrapper/views.py]: Tests were failing on:

```
  File "/mnt/commcare-hq-ro/corehq/apps/receiverwrapper/views.py", line 278, in _secure_post_digest
    authenticated=True,
  File "/mnt/commcare-hq-ro/corehq/apps/receiverwrapper/views.py", line 71, in _process_form
    _record_metrics(metric_tags, 'ignored', response)
  File "/mnt/commcare-hq-ro/corehq/apps/receiverwrapper/views.py", line 164, in _record_metrics
    if result.xform and result.xform.metadata:
AttributeError: 'unicode' object has no attribute 'xform'
```

**Review by emord — APPROVED** [review]

File: corehq/apps/receiverwrapper/views.py [hunk 2]
```diff
@@ -160,7 +160,11 @@ def _submission_error(request, message, count_metric, metric_tags,
     return response
 
 
-def _record_metrics(tags, submission_type, response, timer=None):
+def _record_metrics(tags, submission_type, response, timer=None, xform=None):
+    if xform and xform.metadata:
+        lag = xform.received_on - xform.metadata.timeEnd
```
Comment (dannyroberts) [inline corehq/apps/receiverwrapper/views.py]: This is causing a large number of submission 500s (at least on production): https://sentry.io/organizations/dimagi/issues/1114102925/?project=136860&query=is%3Aunresolved&statsPeriod=14d, https://app.datadoghq.com/dashboard/4fg-n5r-sgn/500-errors?from_ts=1563156475297&live=true&tile_size=m&to_ts=1563329275297&fullscreen_widget=2943715324358578&fullscreen_section=overview

Comment (dannyroberts) [reply corehq/apps/receiverwrapper/views.py]: I think this should fix it: https://github.com/dimagi/commcare-hq/pull/24866

**Pull request merged**